exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -559,191 +848,376 @@ func (e *Exchange) CancelBatchOrders(ctx context.Context, o []order.Cancel) (*or
-	}
-
-	resp, err := e.GetAuthenticatedOrderStatus(ctx, orderID)
-	if err != nil {
-		if len(orderInfo.Trades) > 0 { // on closed orders return trades only
-			if strings.Contains(err.Error(), "Order not found") {
-				orderInfo.Status = order.Closed
-			}
-			return &orderInfo, nil
+		if len(fResults) != 1 {
+			return nil, order.ErrOrderNotFound
 		}
-		return nil, err
-	}
-
-	if orderInfo.Status, err = order.StringToOrderStatus(resp.Status); err != nil {
-		log.Errorf(log.ExchangeSys, "%s %v", e.Name, err)
+		orderDetail := fResults[0]
+		if !pair.IsEmpty() && !orderDetail.Symbol.Equal(pair) {
+			return nil, fmt.Errorf("order with ID %s expected a symbol %v, but got %v", orderID, pair, orderDetail.Symbol)
+		}
+		oType, err := order.StringToOrderType(orderDetail.OrderType)
+		if err != nil {
+			return nil, err
+		}
+		return &order.Detail{
+			Price:                orderDetail.Price.Float64(),
+			Amount:               orderDetail.Quantity.Float64(),
+			AverageExecutedPrice: orderDetail.AveragePrice.Float64(),
+			QuoteAmount:          orderDetail.AveragePrice.Float64() * orderDetail.ExecQuantity.Float64(),
+			ExecutedAmount:       orderDetail.ExecQuantity.Float64(),
+			RemainingAmount:      orderDetail.Quantity.Float64() - orderDetail.ExecQuantity.Float64(),
+			OrderID:              orderDetail.OrderID,
+			Exchange:             e.Name,
+			ClientOrderID:        orderDetail.ClientOrderID,
+			Type:                 oType,
+			Side:                 stringToOrderSide(orderDetail.Side),
+			Status:               orderStateFromString(orderDetail.State),
+			AssetType:            asset.Futures,
+			Date:                 orderDetail.CreationTime.Time(),
+			LastUpdated:          orderDetail.UpdateTime.Time(),
+			Pair:                 orderDetail.Symbol,
+			TimeInForce:          orderDetail.TimeInForce,
+		}, nil
+	default:
+		return nil, fmt.Errorf("%w: %q", asset.ErrNotSupported, assetType)
 	}
-	orderInfo.Price = resp.Rate
-	orderInfo.Amount = resp.Amount
-	orderInfo.Cost = resp.Total
-	orderInfo.Fee = resp.Fee
-	orderInfo.QuoteAmount = resp.StartingAmount
+}
 
-	orderInfo.Side, err = order.StringToOrderSide(resp.Type)
-	if err != nil {
-		return nil, err
+// stringToOrderSide converts order side string representation to order.Side instance
+func stringToOrderSide(orderSide string) order.Side {
+	switch strings.ToUpper(orderSide) {
+	case order.Sell.String():
+		return order.Sell
+	case order.Buy.String():
+		return order.Buy
+	case order.Short.String():
+		return order.Short
+	case order.Long.String():
+		return order.Long
+	default:
+		return order.UnknownSide
 	}
+}
 
-	orderInfo.Date, err = time.Parse(time.DateTime, resp.Date)
-	if err != nil {
-		return nil, err
+// orderStateFromString returns an order.Status instance from a string representation
+func orderStateFromString(orderState string) order.Status {
+	switch orderState {
+	case "NEW":
+		return order.New
+	case "FAILED":
+		return order.Closed
```

Comment:
Rejected?

Reply:
> That has close meaning

Assuming your reply means "FAILED" means `order.Closed`.

Can you cite the source of that information, please?

The documentation doesn't seem to say that "FAILED" means "Closed" anywhere I can find.
Did you find out by testing the API?

It's such a drastically different meaning from what anyone would think "FAILED" would mean, that we need a comment to explain it so maintainers don't constantly think it's wrong.